tests/integration/test_calibrate_mesmer_newcodepath.py

```diff
@@ -86,46 +89,104 @@ def test_calibrate_mesmer(
     REFERENCE_PERIOD = slice("1850", "1900")
 
     HIST_PERIOD = slice("1850", "2014")
-    PROJ_PERIOD = slice("2015", "2100")
 
     LOCALISATION_RADII = range(1750, 2001, 250)
 
     esm = "IPSL-CM6A-LR"
-    scenario = scenarios[0]
     test_cmip_generation = 6
 
     # define paths and load data
     TEST_DATA_PATH = pathlib.Path(test_data_root_dir)
-    TEST_PATH = TEST_DATA_PATH / "output" / "tas" / "one_scen_one_ens"
+    TEST_PATH = TEST_DATA_PATH / "output" / outname
 
     cmip_data_path = (
         TEST_DATA_PATH / "calibrate-coarse-grid" / f"cmip{test_cmip_generation}-ng"
     )
 
-    path_tas = cmip_data_path / "tas" / "ann" / "g025"
+    CMIP_FILEFINDER = FileFinder(
+        path_pattern=cmip_data_path / "{variable}/{time_res}/{resolution}",  # type: ignore
+        file_pattern="{variable}_{time_res}_{model}_{scenario}_{member}_{resolution}.nc",
+    )
+
+    fc_scens = CMIP_FILEFINDER.find_files(
+        variable="tas", scenario=scenarios, model=esm, resolution="g025", time_res="ann"
+    )
+
+    # only get the historical members that are also in the future scenarios, but only once
+    unique_scen_members = fc_scens.df.member.unique()
+
+    fc_hist = CMIP_FILEFINDER.find_files(
+        variable="tas",
+        scenario="historical",
+        model=esm,
+        resolution="g025",
+        time_res="ann",
+        member=unique_scen_members,
+    )
 
-    fN_hist = path_tas / f"tas_ann_{esm}_historical_r1i1p1f1_g025.nc"
-    fN_proj = path_tas / f"tas_ann_{esm}_{scenario}_r1i1p1f1_g025.nc"
+    fc_all = FileContainer(pandas.concat([fc_hist.df, fc_scens.df]))
+
+    scenarios_whist = scenarios.copy()
```

Comment:
```suggestion
    scenarios_incl_hist = scenarios.copy()
```
I think that's easier to understand